### PR TITLE
Fixes #699 - Move content-server to Dockerfile

### LINKED ISSUE #699: Move content-server to a Dockerfile

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: containers/images/content-server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM quay.io/foreman/pulp
+
+RUN echo "tsflags=nodocs" >> /etc/yum.conf && \
+    yum -y install epel-release && \
+    yum -y install python-pulp-streamer && \
```
Comment (ohadlevy) [inline containers/images/content-server/Dockerfile]: you can also rm -rf /var/cache/yum

File: containers/images/content-server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+RUN ln -s /etc/pki/pulp/certs/pulp.crt /etc/pki/tls/certs/localhost.crt
+RUN ln -s /etc/pki/pulp/private/pulp.key /etc/pki/tls/private/localhost.key
+
+RUN chown apache:apache /var/lib/pulp
+
+RUN sed -i 's/127.0.0.1/squid/g' /etc/httpd/conf.d/pulp_streamer.conf
+
+COPY container-assets/content_wsgi.conf /etc/httpd/conf.d/content_wsgi.conf
+
+CMD ["httpd", "-D", "FOREGROUND"]
```
[the remark this answers is not in the document]
Comment (carbonin) [reply containers/images/content-server/Dockerfile]: IIRC httpd does the permission and user change when you start it up, so it should run as root.

[the remark this answers is not in the document]
Comment (ohadlevy) [reply containers/images/content-server/Dockerfile]: right, its needed in order to bind to port 80/443, thanks.

File: containers/images/content-server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM quay.io/foreman/pulp
+
+RUN echo "tsflags=nodocs" >> /etc/yum.conf && \
+    yum -y install epel-release && \
+    yum -y install python-pulp-streamer && \
+		yum clean all
+
+RUN rm -f /etc/httpd/conf.d/pulp.conf /etc/httpd/conf.d/pulp_puppet.conf
```
Comment (ripcurld) [inline containers/images/content-server/Dockerfile]: Each Dockerfile instruction like `RUN` creates a new layer and from what I understood - avoiding adding more layers to the image is a best practice.

Thus, maybe you can merge these next lines into:

```
RUN rm -f /etc/httpd/conf.d/pulp.conf /etc/httpd/conf.d/pulp_puppet.conf /etc/pki/tls/certs/localhost.crt /etc/pki/tls/private/localhost.key && \
    ln -s /var/lib/pulp/published /var/www/pub && \
    ln -s /etc/pki/pulp/certs/pulp.crt /etc/pki/tls/certs/localhost.crt && \
    ln -s /etc/pki/pulp/private/pulp.key /etc/pki/tls/private/localhost.key && \
    chown apache:apache /var/lib/pulp && \
    sed -i 's/127.0.0.1/squid/g' /etc/httpd/conf.d/pulp_streamer.conf
```
Unless there are caching or other concerns that made you decide to do that.

File: containers/images/content-server/Dockerfile (hunk 1 above)
Comment (ehelms) [reply containers/images/content-server/Dockerfile]: I tested that and it didn't seem to do anything more than `yum clean all`.

File: containers/images/content-server/Dockerfile (hunk 3 above)
Comment (ehelms) [reply containers/images/content-server/Dockerfile]: It does reduce layers, however, I think it makes readability worse. I think we can explore image sizes across the board after the initial round to weigh size reduction vs. understandbility.

**Comment by ehelms** [comment]
@ohadlevy @carbonin please re-review at your convience

File: containers/container.yml [hunk 4]
```diff
@@ -76,10 +76,6 @@ services:
     from: centos:7
     roles:
       - squid
-  content-server:
-    from: foreman-pulp-base:latest
-    roles:
-      - content-server
```
Comment (carbonin) [inline containers/container.yml]: Same as before, can we remove this role?

Comment (ehelms) [reply containers/container.yml]: With fire -- done

Comment (carbonin) [reply containers/container.yml]: :fire: :fire: :scissors: :+1:

**Pull request merged**